test/StripTagsTest.php

```diff
@@ -511,6 +511,28 @@ public function testMultiQuoteInput()
         $this->assertEquals($expected, $filter->filter($input));
     }
 
+    public function badCommentProvider()
+    {
+        return [
+            ['A <!--> B', 'A '],  // Should be treated as just an open
```

Comment:
I love the tests. But just curious, would tests about these scenarios make sense as well:
- multiline comments
- stacked comments (i know this seems odd, but still it should work `A <!-- <!-- --> --> B` should return `A B`

Reply:
@TotalWipeOut would you be able to add scenarios mentioned above?

I would suggest also: `A <!--> B <!--> C` and I believe it should return `A   C`.

Reply:
@icanhazstring

case with nested comment is invalid (it is not possible to have nested comments).
I agree that we should decide somehow to proceed them, and I would go with the same as modern browser. Just checked the following example on Chrome:

```
A <!-- B <!-- C --> D --> E
```

and as the result I've got:
```
A D --> E
```

so not `A   E` as you would expect.

Reply:
Yes, I will add these extra tests

Reply:
@webimpress and @icanhazstring We have an issue with this test:
`A <!-- B <!-- C --> D --> E` -> `A D --> E`
After the loop for stripping comments, `$value` is correctly `A  --> B`
The following loop strips the lone `>` so the result is `A  -- B` i would guess this is a different bug?

Reply:
@TotalWipeOut I've checked it, and it looks like desired behaviour. We have tests to cover these scenarios, that `>` is removed, for example:
`testFilterGt`: `Ensures that any greater-than symbols ‘>’ are removed from text having no tags`:

https://github.com/zendframework/zend-filter/blob/922fe11b1595f681ef9618f630ec090070bbf53d/test/StripTagsTest.php#L200-L211

Reply:
@webimpress totally fine with me to match the filter with modern browser behavior.

Reply:
@webimpress I agree, it was designed to strip orphan `>` characters. But I would say that this isn't correct behaviour, considering that `strip_tags()` and Chrome leave that character untouched.
@icanhazstring If we do that, a lot of tests will fail and need to be re-written.

How do I proceed?

Reply:
@TotalWipeOut I would keep it for now, as changing it now will be BC Break. We should log another issue and change the behaviour in next major version. So for now - in your test - expected value should be `A  D -- E` as you said.

Reply:
OK, thanks. I have added the requested tests, 11 in total now :slightly_smiling_face: